server/models/product.js

```diff
@@ -63,4 +64,177 @@ const productSchema = new Schema({
 
 userSchema.plugin(uniqueValidator);
 
-module.exports = mongoose.model('Product', productSchema);
+const Product = mongoose.model('Product', productSchema);
+
+/**
```

Comment:
Thank you so much! I understood, I need to separate logic from the API handler (which is only responsible for validating input data and delegating tasks to our logic file). It will be better to put these logic codes into a separate file (called productService), right?

Reply:
You can either do `ProductService` or simply doing it inside your models for now.

Then separate into another service layer when there are more logic. Right now, it seems a little overkill because most of your business logic is query and update data.

Reply:
I see. I have a question: Apart from querying and updating data, what kind of logic looks like makes us separate it into a service layer?

Reply:
Probably not much for this app, but Analytics would be a possibility. For example if you need to present a stock inventory graph over time (like how much stock you keep over time), that information may not be expressed as a simple query.

Most real world systems out there generally have a logic layer. E-commerce platforms would have services to show performance of your store, and maybe even recommendation on how to improve your sales. Those are more complex than data query.

Reply:
I understood! Thank you :)